powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/PowerAuthEncryptionProviderBase.java

```diff
@@ -64,17 +71,35 @@ public abstract class PowerAuthEncryptionProviderBase {
 
     private final ObjectMapper objectMapper = new ObjectMapper();
     private final EciesFactory eciesFactory = new EciesFactory();
+    private final KeyGenerator keyGenerator = new KeyGenerator();
 
     /**
      * Get ECIES decryptor parameters from PowerAuth server.
      *
      * @param activationId Activation ID (only used in activation scope, in application scope use null).
      * @param applicationKey Application key.
      * @param ephemeralPublicKey Ephemeral public key for ECIES.
+     * @param version ECIES protocol version.
+     * @param nonce ECIES nonce.
+     * @param timestamp Timestamp for ECIES.
      * @return ECIES decryptor parameters.
      * @throws PowerAuthEncryptionException In case PowerAuth server call fails.
      */
-    public abstract @Nonnull PowerAuthEciesDecryptorParameters getEciesDecryptorParameters(@Nullable String activationId, @Nonnull String applicationKey, @Nonnull String ephemeralPublicKey) throws PowerAuthEncryptionException;
+    public abstract @Nonnull PowerAuthEciesDecryptorParameters getEciesDecryptorParameters(@Nullable String activationId, @Nonnull String applicationKey, @Nonnull String ephemeralPublicKey, @Nonnull String version, String nonce, Long timestamp) throws PowerAuthEncryptionException;
```

Comment:
Just an idea to consider. The list of parameters is quite extensive and a context object could be nicer. For example `GetEciesDecryptorRequest` or something similar.

Reply:
Let's raise an issue, this change has no impact on the functionality. Similar issue may be present on multiple places.